tool/util/path.dart

```diff
@@ -0,0 +1,78 @@
+
+
+String fromUri(uri) {
+  if (uri is String) uri = Uri.parse(uri);
+  return Uri.decodeComponent((uri as Uri).path);
+}
+
+String dirname(String path) {
+  var pt = path.trim();
```

Comment:
Yes, but this script isn't run with pub at all, just executed like a shell script so the entire build would need to be reorganized to make pub packages work.

Initially it was in the path package in my first version of the script on just building one package. It wasn't until I tried to add it to this script that I realized pub support wasn't included. That's why I chose to implement this sub-section of path package the same way it's implemented in the package